lib/IRGen/GenClass.cpp

```diff
@@ -71,13 +71,19 @@ namespace {
   /// Layout information for class types.
   class ClassTypeInfo : public HeapTypeInfo<ClassTypeInfo> {
     ClassDecl *TheClass;
-    mutable Optional<ClassLayout> FieldLayout;
+
+    // First element is the standard layout, which is always available.
+    //
+    // Second element is the completely fragile layout for Objective-C
+    // interop.
+    mutable Optional<ClassLayout> FieldLayout[2];
```

Comment:
I think it makes more sense just to have two variables. It's not like you'll ever use this as an array, and it's just as easy to have your ternary choose between two variables as between two indexes.

Reply:
Fair enough.